src/main/webapp/scripts/poll.js

```diff
@@ -0,0 +1,77 @@
+ /** 
+  * The Poll class checks for fresh data over a given interval by
+  * periodically making API requests to a server. The Poll class can also
+  * be used to periodically update the server of necessary updates. 
+  * By default, polling objects poll indefinitely over a
+  * period of 30 seconds.
+  * 
+  * For example, if a fetchAPIRequest is one that contacts a servlet
+  * to doGet some value, a Poll object will repeatedly doGet the value, 
+  * attaching it to one of its fields which can then be retrieved with 
+  * getResult(). The core nature is that it executes a function over
+  * and over, and doesn't necessarily have to be used to doGet some value,
+  * but can doPost, updating the server also.
+  */ 
+ class Poll {
+  /**
+   * Initializes a Poll object.
+   * @param {function(): ?Object} fetchAPIRequest Represents the API
+   *    request that is polled.
+   * @param {number=} [pollingPeriod] Represents the cadence at which 
+   *    the fetchRequest is called upon. Must be provided in milliseconds
+   *    and is by default 30,000 milliseconds (or 30 seconds). 
+   * @param {?number=} [maxAttempts] An optional parameter, 
+   *    can limit the amount of times polling occurs by the number
+   *    of maxAttempts. By default, there is no limit (defined as null).
+   */
+  constructor(fetchAPIRequest, pollingPeriod = 30000, maxAttempts = null) {
```

Comment:
Rather than have maxAttempts in the constructor, since it's a test only thing, make a setter function like this:
testOnlySetMaxPollAttempts();